core/amber/src/main/python/core/architecture/packaging/input_manager.py

```diff
@@ -119,14 +143,11 @@ def process_data_payload(
             )
 
             if port_completed:
-                yield InputExhausted()
+                yield EndOfUpstream()  # EndOfInputPort()
 
             all_ports_completed = all(
                 map(lambda port: port.is_completed(), self._ports.values())
             )
 
             if all_ports_completed:
-                yield EndOfAll()
-
-        else:
-            raise NotImplementedError()
```

Comment:
keep the else case?

Reply:
It's moved to line 115:
https://github.com/Texera/texera/pull/2802/files#diff-464c3d12c346df3eb757bf393b093d1b2a03e95b07348a533ae44b6f476dfc92R115